app/views/markets/_form.html.erb

```diff
@@ -0,0 +1,15 @@
+<%= form_for item do |f| %>
+  <%= f.label :name, "#{name_desc}" %>
+  <%= f.text_field :name, class: "form_field", required: true, autofocus: true, placeholder: name_placeholder %>
```

Comment:
Nice use of placeholders so that you can supply different values for new and edit.